### PR TITLE
codeintel: Make some oobmigrations permanent jobs

### PR DESCRIPTION
**Pull request by efritz**
This PR makes progress towards #38053. This PR moves two out of band migrations into a permanent background worker routine, as they do not need to be deprecated and will remain useful. They are:

1. An out of band migration that denormalizes the number of references _to_ an upload on that commit record. Will now periodically update batches of records with a NULL `num_references` count, which is used for expiration logic. We will likely want to have a "flush and recalculate" mode this will be useful for in the future.
1. An out of band migration that checks fills the `committed_at` column of completed uploads that do not have the value set. This is mainly to backfill records before we tracked this, but this might be useful for force-pushed commits (or if we ever drop it from the correlation step).

## Test plan

New unit tests on service and store layers.

### CODE REVIEW

File: internal/codeintel/uploads/background/backfill/config.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+	Interval  time.Duration
+	BatchSize int
+}
+
+var ConfigInst = &config{}
+
+func (c *config) Load() {
+	c.Interval = c.GetInterval("CODEINTEL_UPLOAD_BACKFILLER_INTERVAL", "1m", "The frequency with which to run periodic codeintel backfiller tasks.")
+	c.BatchSize = c.GetInt("CODEINTEL_UPLOAD_BACKFILLER_BATCH_SIZE", "25", "The number of upload records to backfill per batch.")
```
Comment (efritz) [inline internal/codeintel/uploads/background/backfill/config.go:20]:
```
c.BatchSize = c.GetInt("CODEINTEL_UPLOAD_BACKFILLER_BATCH_SIZE", "100", "The number of upload to populate an unset `commited_at` field per batch.")
```

File: internal/codeintel/uploads/background/expiration/config.go [hunk 2]
```diff
@@ -9,15 +9,17 @@ import (
...
-	PolicyBatchSize        int
-	RepositoryBatchSize    int
-	RepositoryProcessDelay time.Duration
-	UploadBatchSize        int
-	UploadProcessDelay     time.Duration
+	BranchesCacheMaxKeys           int
+	CleanupTaskInterval            time.Duration
+	CommitBatchSize                int
+	Interval                       time.Duration
+	Interval2                      time.Duration
```
Comment (efritz) [inline internal/codeintel/uploads/background/expiration/config.go:16]: TODO - rename

File: internal/codeintel/uploads/background/expiration/reference_count_updater.go [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+	_ goroutine.Handler      = &referenceCountUpdater{}
+	_ goroutine.ErrorHandler = &referenceCountUpdater{}
+)
+
+func (u *referenceCountUpdater) Handle(ctx context.Context) error {
+	return u.uploadSvc.BackfillReferenceCountBatch(ctx, u.batchSize)
+}
+
+func (u *referenceCountUpdater) HandleError(err error) {
+}
```
Comment (eseliger) [inline internal/codeintel/uploads/background/expiration/reference_count_updater.go:24]: is this intentionally empty?

**Review by eseliger — APPROVED** [review]

File: internal/codeintel/uploads/background/expiration/reference_count_updater.go (hunk 3 above)
Comment (efritz) [reply internal/codeintel/uploads/background/expiration/reference_count_updater.go:24]: Yes, for now

**Pull request merged**